src/Elcodi/Component/Product/Entity/Variant.php

```diff
@@ -80,6 +80,16 @@ class Variant implements VariantInterface
     protected $options;
 
     /**
+     * Gets name
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->product->getName();
```

Comment:
it looks like the proper solution is to move the `getName` and `setName` to `PurchasableInterface`. This will address the previous issue as well. 
A `PurchasableInterface` object is basically a [SKU](https://en.wikipedia.org/wiki/Stock_keeping_unit). I think is up to us if we want to move `name`, `description` and such to the interface. It sounds like it makes sense though.